hack/make-rules/verify/tslint.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this script runs tslint against the repo using docker
+#
+# TODO: we should be using eslint instead now, but this repo is also
+# on a rather old version of node, first we get the build system migrated.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../../.." && pwd -P)"
+cd "${REPO_ROOT}"
+
+# ensure deps are installed
+# TODO: factor out into common script / step?
+hack/run-in-node-container.sh \
+    yarn install
```

Comment:
this takes maybe 2s (some variability), the whole `tslint.sh` takes ~6s total. not great, but hardly the slowest thing we're doing. 

probably not worth the complexity trying to sometimes skip this?
we already have the yarn lock and `node_modules/` cache

Reply:
2s should be fine, we do have many other stuffs that take much longer

Reply:
it will be something like this *per* typescript target though, unless we split it out and use a make dependency or something.

right now it will avoid re-downloading everything, but 2s is a bit slower than I'd like to do at the beginning of each typescript related script/target.

Reply:
I think it depends on whether anyone would do this locally or not, maybe `make verify-tslint`, doubt anyone would need to compile deck locally very frequently.

For prow jobs, I think `lint` job does it once, `integration-test` compiles deck once, and `push-prow-image` also compiles deck once, each with 2s overhead.

so we might be fine?

Reply:
it's also something we can improve on later, factoring it out into a seperate script will make that relatively easy to make a make target dependency later